system/lib/pthread/library_pthread.c

```diff
@@ -763,4 +763,61 @@ uint64_t __atomic_fetch_xor_8(void *ptr, uint64_t value, int memmodel)
 	return _emscripten_atomic_fetch_and_xor_u64(ptr, value);
 }
 
+typedef struct main_args
+{
+  int argc;
+  char **argv;
+} main_args;
+
+extern int __call_main(int argc, char **argv);
+
+void *__emscripten_thread_main(void *param)
+{
+  emscripten_set_thread_name(pthread_self(), "Application main thread"); // This is the main runtime thread for the application.
+  main_args *args = (main_args*)param;
+  return (void*)__call_main(args->argc, args->argv);
+}
+
+static volatile main_args _main_arguments;
```

Comment:
volatile?

Reply:
lgtm. just curious why this is volatile.

Reply:
Agree that it is not needed here from compiler perspective. In general when I have shared state, if I don't use atomic operations to access the data and if there does not exist a mutex object associated with the shared data, I like to mark global data `volatile` more for documentation reasons ("see this thing here is shared between threads") (shared, not to mean synchronized for the pedantics). Here `pthread_create` is the synchronization primitive, rather than an atomic op or a mutex lock.

We don't really have a convention here in the codebase for this yet, could certainly go either way here. wdyt?

Reply:
Got it, sounds fine the current way.